frontend/src/features/admin-form/create/builder-and-design/useFieldBuilderStore.tsx

```diff
@@ -11,64 +11,116 @@ export enum FieldBuilderState {
   Inactive,
 }
 
+type FieldBuilderCreateEditStateData =
+  | {
+      state: FieldBuilderState.CreatingField
+      field: FieldCreateDto
+      insertionIndex: number
+    }
+  | {
+      state: FieldBuilderState.EditingField
+      field: FormFieldDto
+    }
+  | { state: FieldBuilderState.EditingEndPage }
+
 export type FieldBuilderStore = {
-  updateCreateState: (field: FieldCreateDto, insertionIndex: number) => void
-  updateEditState: (field: FormFieldDto) => void
-  setEditEndPage: () => void
-  setToInactive: () => void
+  updateCreateState: (
+    field: FieldCreateDto,
+    insertionIndex: number,
+    holding?: boolean,
+  ) => void
+  updateEditState: (field: FormFieldDto, holding?: boolean) => void
+  setEditEndPage: (holding?: boolean) => void
+  setToInactive: (holding?: boolean) => void
+  isDirty: boolean
+  setIsDirty: (isDirty: boolean) => void
```

Comment:
since these are used throughout design and endpage too, would it be better to put them somewhere above, perhaps in the Create page context or Admin builder context? Because the original intention for separating the design, builder and endpage stores was to separate the concerns, thus putting `isDirty` here feels a bit counter to that as its also used in design and endpage

Reply:
didn't want to put something that changes so often into context, since that will trigger rerenders. Opted to put in this store first, we can always move it in the future. Seemed overkill to add yet another store just to hold one dirty state. do you think we should create a new store? actually ok let me do it

Reply:
done in 4ecda0b